Add OCI-Referrers header (#463)
opencontainers/distribution-spec: spec.md

See [Content Digests](https://github.com/opencontainers/image-spec/blob/v1.0.1/descriptor.md#digests) <sup>[apdx-3](#appendix)</sup> for information on how to detect the hashing algorithm in use.
Most clients MAY ignore the value, but if it is used, the client MUST verify the value against the uploaded blob data.

> A registry that implements the [Referrers Listing](#listing-referrers) MAY include the `OCI-Referrers: absent` header on manifests which do not have any referrers.

Reviewer: ```suggestion
A registry that implements the [Referrers Listing](#listing-referrers) MAY include the `OCI-Referrers: absent` or `OCI-Referrers: present` headers on manifests to indicate whether they, presently, have any referrers.
```

> Clients MAY skip requests to the [Referrers Listing](#listing-referrers) for manifests that include the `OCI-Referrers: absent` header.

Reviewer: ```suggestion
Clients should recognize that receipt of the `OCI-Referrers: absent` or  `OCI-Referrers: present` headers, are a point in time notice. At any time, manifests with the subject references that cause the notice may be removed or added after the notice has been provided.
```
